txpipe/spatial_diagnostics.py

```diff
@@ -0,0 +1,75 @@
+from .base_stage import PipelineStage
+from .data_types import Directory, HDFFile, PNGFile
+import numpy as np
+
+class TXFocalPlanePlot(PipelineStage):
+    """
+    Make diagnostic plot of 
+    mean measured ellipticity and residual
+    as a function of position in the focal plane
+    """
+
+    name = "TXFocalPlanePlot"
+
+    inputs = [
+        ("star_catalog", HDFFile),
+    ]
+
+    outputs = [
+        ("focalplane_g", PNGFile),
+    ]
+
+    config_options = {}
+
+    def run(self):
+        import matplotlib.pyplot as plt
+        
+        
+        fov_x, fov_y, e1, e2, de1, de2 = self.load_stars()
+        
+        X = plt.hist2d(fov_x, fov_y, bins=(100,100))
+        
+        x_edge = X[1]
+        y_edge = X[2]
+        
+        PSF_e1_err = np.zeros((100,100))
+        PSF_e2_err = np.zeros((100,100))
+
+        for i in range(100):
+            for j in range(100):
+                mask = (fov_x>=x_edge[i])*(fov_x<x_edge[i+1])*(fov_y>=y_edge[j])*(fov_y<y_edge[j+1])
+                if len(e1[mask])>0:
+                    PSF_e1_err[i][j] = np.mean(de1[mask])
```

Comment:
It looks like you don't actually use these values - is that intended?

You might have some left over code here.

Reply:
deleted!